docs/scripts/generate-test-runner-guest-matrix.py

```
@@ -46,24 +46,24 @@ def main() -> None:
             re.compile(pattern) for pattern in definitions['unsupported']
         ]
 
-    except re.error as exc:
-        raise GeneralError(f"Invalid 'unsupported' pattern '{exc.pattern}'.") from exc
+    except re.error as error:
+        raise GeneralError("Invalid 'unsupported' pattern.") from error
```

TBH, I for one would keep this as it was. There is a difference between `raise GeneralError(f"Invalid 'unsupported' pattern '{exc.pattern}'.") from error`, which shows the problematic bit right in the message - and just the relevant bit, and let's say `raise GeneralError(f"Invalid 'unsupported' pattern: {exc}'.") from exc`, where the whole `exc` message is bundled into our message. The first is good: the important piece of information, that the patter nis broken and *which* pattern it was, is right in the message, with `raise ... from ...` providing the detailed chain if needed. The second is something we want to avoid, i.e. bundling the whole exception's `str()` into our message.

The rule is "don't bundle whole exceptions into our messages", not necessarily "don't bundle anything from raised exceptions into our messages" - in this case, for example, it's very useful to pick the pattern and mention it in our message. Without it, user would need to the cause, and look it up there.